frontend/src/Dashboard/HistoricData.js

```diff
@@ -0,0 +1,283 @@
+import React, { useState, useEffect } from 'react';
+import MeasureCard from '../MeasureCard/MeasureCard.js';
+import axios from 'axios';
+
+import { Swiper, SwiperSlide } from 'swiper/react';
+import SwiperCore, { Navigation, Pagination, EffectCoverflow } from 'swiper';
+
+import { IconContext } from 'react-icons';
+import {
+  RiCompassDiscoverFill,
+  RiDashboard3Line,
+  RiLeafLine,
+  RiBattery2ChargeLine,
+} from 'react-icons/ri';
+import { IoMdRocket } from 'react-icons/io';
+import { WiHumidity, WiThermometer } from 'react-icons/wi';
+import { HiSun } from 'react-icons/hi';
+import { TiWeatherShower, TiWeatherSnow } from 'react-icons/ti';
+import { VscDashboard } from 'react-icons/vsc';
+import { GiDrop, GiChaliceDrops, GiCarBattery } from 'react-icons/gi';
+
+import 'swiper/swiper.scss';
+import './SwiperDesign.css';
+import 'swiper/components/pagination/pagination.scss';
+import 'swiper/components/effect-coverflow/effect-coverflow.scss';
+import 'swiper/components/navigation/navigation.scss';
+
+SwiperCore.use([Navigation, Pagination, EffectCoverflow]);
+
+export default function HistoricData() {
+  const [weatherData, setWeatherData] = useState([]);
+
+  useEffect(() => {
+    async function fetchData() {
+      const response = await axios.get('/api/test/12');
+      response.data && setWeatherData(formatData(response.data));
```

Comment:
A formatData-t ne a state setelés előtt hívd meg, hanem render közben, tehát itt csak seteld a választ a state-be, lentebb pedig formatData(weatherData).map ...